app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -593,4 +608,37 @@ mod tests {
         assert_eq!(group_at_1.name, "Group 1");
         check_names_and_order_of_group_entries(group_at_1, &[qn_of_db_entry_1.name()]);
     }
+
+    /// Test building a component list with a private component. The private component will be
+    /// excluded from the list.
+    #[test]
+    fn building_component_list_with_private_component() {
+        let suggestion_db = enso_suggestion_database::mock_suggestion_database! {
+            test.Test {
+                mod TopModule_1 {
+                    fn fun_2() -> Standard.Base.Any;
+                    #[with_doc_section(enso_suggestion_database::doc_section!(@ "PRIVATE", ""))]
+                    fn fun_3() -> Standard.Base.Any;
+                    fn fun_4() -> Standard.Base.Any;
+                }
+            }
+        };
+        let mut builder = List::new().with_local_scope_module_id(1);
+        let entry_ids = 0..5;
```

Comment:
I see only `fun_2`, `fun_3`, and `fun_4` refs. Where does other entries come from?

Reply:
> I see only `fun_2`, `fun_3`, and `fun_4` refs. Where does other entries come from?

`test.Test` and `mod TopModule_1` also are suggestion database entries.